src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/AccountsReferenceDateQuestionController.java

```diff
@@ -0,0 +1,127 @@
+package uk.gov.companieshouse.web.accounts.controller.smallfull;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.model.accounts.smallfull.SmallFullApi;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.api.ApiClientService;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.AccountsReferenceDateQuestion;
+import uk.gov.companieshouse.web.accounts.model.state.CompanyAccountsDataState;
+import uk.gov.companieshouse.web.accounts.service.company.CompanyService;
+import uk.gov.companieshouse.web.accounts.service.smallfull.impl.SmallFullServiceImpl;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.time.LocalDate;
+
+@Controller
+@NextController(DirectorsReportQuestionController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/small-full/accounts-reference-date-question")
+public class AccountsReferenceDateQuestionController extends BaseController {
+
+    @Autowired
+    private CompanyService companyService;
+
+    @Autowired
+    private SmallFullServiceImpl smallFullService;
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    private static final String ACCOUNTS_REFERENCE_DATE_QUESTION = "accountsReferenceDateQuestion";
+
+    @GetMapping
+    public String getAccountsReferenceDateQuestion(@PathVariable String companyNumber,
+                                                   @PathVariable String transactionId,
+                                                   @PathVariable String companyAccountsId,
+                                                   Model model,
+                                                   HttpServletRequest request) {
+
+        CompanyProfileApi companyProfile;
+        SmallFullApi smallFullAccounts;
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        AccountsReferenceDateQuestion accountsReferenceDateQuestion = new AccountsReferenceDateQuestion();
+
+        try {
+            companyProfile = companyService.getCompanyProfile(companyNumber);
```

Comment:
Nitpick: maybe we could nest all of the logic within the `try` block? Granted it doesn't buy us much and is more stylistic preference, however we currently initialise variables outside of the try and instantiate them within. If we nested all of the logic within the try clause, we wouldn't need to do that

Reply:
moved into try block, looks clean so I'm also happy with re-structure